partiql-ast/src/main/kotlin/org/partiql/ast/sql/SqlDialect.kt

```diff
@@ -224,9 +224,11 @@ public abstract class SqlDialect : AstBaseVisitor<SqlBlock, SqlBlock>() {
         return head concat r(value)
     }
 
-    override fun visitExprIon(node: Expr.Ion, head: SqlBlock): SqlBlock {
-        // simplified Ion value writing, as this intentionally omits formatting
-        val value = node.value.toString()
+    override fun visitExprVariant(node: Expr.Variant, head: SqlBlock): SqlBlock {
+        if (node.encoding != "ion") {
```

Comment:
Maybe having the encoding as an `enum` is a better option.

Reply:
Agreed, however I'm not sure the best way to make it extensible/extendable

Reply:
I was thinking of it like the "content-type" header

Reply:
I see thinking of it same as "content-type" make sense if we are directly working on the serialized payload. But if the node is an in-memory (which can be deserialized from the payload), we may benefit from Enum.